lib/network.py

```diff
@@ -556,11 +557,12 @@ def switch_to_interface(self, server, switch_reason=None):
 
     def close_interface(self, interface):
         if interface:
```

Comment:
Better to do this in reverse here? 

```
with self.interface_lock:
    if interface:
        do stuff
```

Instead of
```
if interface
    with self.interface_lock:
        do stuff
```

Reply:
Technically, true.  I'll fix it.  (Although this code is still so riddled with race conditions it's not like it matters much).

Reply:
Wait.. that's what I had.  Not sure what you mean to be honest..

Care to explain?

Reply:
Oh nevermind.  I see it. Github is weird.

Reply:
Ok so I opened op the code and: Actually.. it's a passed-in variable that is a reference to something on the conceptual "stack" -- so no.. it doesn't need the lock before checking it.  If it were `self.interface` it would. Leaving it as it is. :)